ingest-app/src/cmr/ingest/api/provider.clj

```diff
@@ -126,14 +147,20 @@
       (acl/verify-ingest-management-permission request-context :update)
       (common-enabled/validate-write-enabled request-context "ingest")
       (one-result->response-map
-        (provider-service/create-provider request-context
-                                          (validate-and-prepare-provider-concept
-                                           (read-body headers body)))))
+       (provider-service/create-provider request-context
+                                         (validate-and-prepare-provider-concept
+                                          (read-body headers body)))))
 
     ;; read a provider
     (GET "/:provider-id" {{:keys [provider-id]} :params
                           request-context :request-context}
-      (one-result->response-map (provider-service/read-provider request-context provider-id)))
+      (let [response (provider-service/read-provider request-context provider-id)]
+        (try
+          (acl/verify-ingest-management-permission request-context :read)
+          (result->response-map response)
+          (catch clojure.lang.ExceptionInfo _
```

Comment:
any exception?

Reply:
I changed this to also include a check for the expected message response to make sure I don't act on other errors.